kobo/apps/stripe/views.py

```diff
@@ -1,12 +1,137 @@
+    CheckoutLinkSerializer,
+    CustomerPortalSerializer,
+    ProductSerializer,
+)
+
+from kobo.apps.organizations.models import (
+    OrganizationUser,
+    OrganizationOwner,
+    Organization
+)
+
+class CheckoutLinkView(
+    APIView
+):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CheckoutLinkSerializer
+
+    @staticmethod
+    def generate_payment_link(price_id, user, organization_uid):
+        if organization_uid:
+            # Look up the specific organization provided
+            try:
+                organization_user = OrganizationUser.objects.get(user=user)
+                organization = Organization.objects.get(
+                    uid=organization_uid
+                )
+            # If we can't find the organization or organization user, bail
+            except ObjectDoesNotExist:
+                return Response(
+                    {'status': r'Organization does not exist'}, status=status.HTTP_403_FORBIDDEN
+                )
+        else:
+            # find the organization the user belongs to, otherwise make a new one
+            organization_user, _ = OrganizationUser.objects.get_or_create(user=user)
+            organization, _ = Organization.objects.get_or_create(users=user)
+        # Make sure the organization owner is the same as the logged-in user
+        if organization.owner.id != organization_user.id:
+            return Response({'status': r'Logged-in user is not organization owner'}, status=status.HTTP_403_FORBIDDEN)
+        customer, created = Customer.get_or_create(
+            subscriber=organization,
+            livemode=settings.STRIPE_LIVE_MODE
+        )
+        session = CheckoutLinkView.start_checkout_session(customer.id, price_id, organization.uid)
+        return Response({'url': session})
+
+    @staticmethod
+    def start_checkout_session(customer_id, price_id, organization_uid):
+        session = stripe.checkout.Session.create(
+            api_key=djstripe_settings.STRIPE_SECRET_KEY,
+            automatic_tax={
+                'enabled': False
+            },
+            customer=customer_id,
+            line_items=[
+                {
+                    "price": price_id,
+                    "quantity": 1,
+                },
+            ],
+            metadata={
+                'organization_uid': organization_uid
+            },
+            mode="subscription",
+            payment_method_types=["card"],
+            success_url=f'{settings.KOBOFORM_URL}/#/plans?checkout_complete=true',
+        )
+        return session
+
+    def post(self, request):
+        serializer = CheckoutLinkSerializer(data=request.query_params)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        serializer_data = serializer.validated_data
+        price_id = serializer_data['price_id']
+        try:
+            organization_uid = serializer.validated_data['organization_uid']
```

Comment:
Nit-pick, but I would write this as `serializer.validated_data.get('organization_uid'). 1 line vs 4 and just as readable.